src/overlay/test/OverlayTests.cpp

```diff
@@ -464,7 +464,8 @@ TEST_CASE("outbound queue filtering", "[overlay][connections]")
             constructSCPMsg(env);
             peer->addMsgAndMaybeTrimQueue(constructSCPMsg(env));
         }
-        REQUIRE(scpQueue.empty());
+        // Always keep checkpoint messages
```

Comment:
Could we also test that this queue properly purges checkpoints ledgers (so scpQueue is empty at the end)? Should be a tiny change, like setting `env.statement.slotIndex` to something really old.